src/exchange_adapters/alphavantage.ts

```diff
@@ -0,0 +1,142 @@
+   *  {
+   *    "Realtime Currency Exchange Rate": {
+   *      "1. From_Currency Code": "XOF",
+   *      "2. From_Currency Name": "CFA Franc BCEAO",
+   *      "3. To_Currency Code": "EUR",
+   *      "4. To_Currency Name": "Euro",
+   *      "5. Exchange Rate": "0.00153000",
+   *      "6. Last Refreshed": "2023-08-03 07:41:09",
+   *      "7. Time Zone": "UTC",
+   *      "8. Bid Price": "0.00152900",
+   *      "9. Ask Price": "0.00153000"
+   *    }
+   *  }
+   */
+  parseTicker(json: Response): Ticker {
+    const response = json['Realtime Currency Exchange Rate']
+
+    const from = response[ResponseKeys.fromCurrency]
+    const to = response[ResponseKeys.toCurrency]
+    assert(
+      from === this.config.baseCurrency,
+      `From currency mismatch in response: ${from} != ${this.config.baseCurrency}`
+    )
+    assert(
+      to === this.config.quoteCurrency,
+      `To currency mismatch in response: ${to} != ${this.config.quoteCurrency}`
+    )
+
+    const ticker = {
+      ...this.priceObjectMetadata,
+      ask: this.safeBigNumberParse(response[ResponseKeys.ask])!,
+      bid: this.safeBigNumberParse(response[ResponseKeys.bid])!,
+      lastPrice: this.safeBigNumberParse(response[ResponseKeys.rate])!,
+      timestamp: this.toUnixTimestamp(response[ResponseKeys.lastUpdated]),
+      // These FX API's do not provide volume data,
+      // therefore we set all of them to 1 to weight them equally
+      baseVolume: new BigNumber(1),
+      quoteVolume: new BigNumber(1),
+    }
+    this.verifyTicker(ticker)
+    return ticker
+  }
+
+  toUnixTimestamp(datetime: string): number {
+    // datetime format: YYYY-MM-DD HH:MM:SS
+    const date = datetime.split(' ')[0]
+    const time = datetime.split(' ')[1]
+
+    const [year, month, day] = date.split('-').map(Number)
+    const [hours, minutes, seconds] = time.split(':').map(Number)
+
+    // month should be 0-indexed
+    return Date.UTC(year, month - 1, day, hours, minutes, seconds) / 1000
+  }
+
+  async isOrderbookLive(): Promise<boolean> {
+    /*
+      This check requires us to make a call to the ticker endpoint, however we can't
+      use fetchTicker() because it creates a circular dependency: 
+      fetchTicker requires isOrderbookLive under the hood, which requires fetchTicker, etc.
+
+      Therefore we make a similar call using fetchFromApi() and the orderbook status type.
+
+      We will have the same scenario for other adapters as well so we can find a nicer 
+      way to do this in the future. 
+    */
+    const base = this.config.baseCurrency
+    const quote = this.config.quoteCurrency
+
+    const tickerJson: Response = await this.fetchFromApi(
+      ExchangeDataType.ORDERBOOK_STATUS,
+      `query?function=CURRENCY_EXCHANGE_RATE&from_currency=${base}&to_currency=${quote}&apikey=${this.config.apiKey}`
+    )
+    const ticker = this.parseTicker(tickerJson)
+
+    /*
+      TODO: verify the behaviour of the ticker endpoit on the weekend.
```

Comment:
do we have an update on that?

Reply:
Yeah I checked it on Sunday evening and the timestamp was still getting updated, which is not great. This is the response from Sunday 6th ~22:55 CET time:

```
{
    "Realtime Currency Exchange Rate": {
        "1. From_Currency Code": "XOF",
        "2. From_Currency Name": "CFA Franc BCEAO",
        "3. To_Currency Code": "EUR",
        "4. To_Currency Name": "Euro",
        "5. Exchange Rate": "0.00153000",
        "6. Last Refreshed": "2023-08-06 20:54:04",
        "7. Time Zone": "UTC",
        "8. Bid Price": "0.00152900",
        "9. Ask Price": "0.00153000"
    }
}
```

So based on that this check wouldn't work. However as we discussed yesterday, this is just for testing purposes so we'll roll with it for now and decide later on how do we want to approach this.